src/seedsigner/views/tools_views.py

```diff
@@ -256,6 +264,194 @@ def run(self):
+
+        if selected_menu_num == RET_CODE__BACK_BUTTON:
+            return Destination(ToolsCoinEntropyMnemonicLengthView)
+
+        selected_option = button_data[selected_menu_num].return_data
+
+        if selected_option == "all":
+            return Destination(ToolsCoinEntropyEntryView, view_args={"total_flips": self.total_flips})
+
+        elif selected_option == "setwise":
+            return Destination(ToolsCoinEntropySetwiseEntryView, view_args={"total_flips": self.total_flips})
+
+
+
+class ToolsCoinEntropyEntryView(View):
+    # This View handles the "all flips in one go" option for coin flip entropy input,
+    # where the user enters all required coin flips (128 or 256) in a single session.
+    def __init__(self, total_flips: int):
+        super().__init__()
+        self.total_flips = total_flips
+    
+
+    def run(self):
+        ret = self.run_screen(
+            ToolsCoinEntropyEntryScreen,
+            return_after_n_chars=self.total_flips,
+        )
+
+        if ret == RET_CODE__BACK_BUTTON:
+            return Destination(BackStackView)
+
+        coin_seed_phrase = mnemonic_generation.generate_mnemonic_from_coin_flips(ret)
+
+        # Add the mnemonic as an in-memory Seed
+        seed = Seed(coin_seed_phrase, wordlist_language_code=self.settings.get_value(SettingsConstants.SETTING__WORDLIST_LANGUAGE))
+        self.controller.storage.set_pending_seed(seed)
+
+        # Cannot return BACK to this View
+        return Destination(SeedWordsWarningView, view_args={"seed_num": None}, clear_history=True)
+
+
+
+class ToolsCoinEntropySetwiseEntryView(View):
+    """ Handles set-wise coin flip entry (11-bit groups for BIP-39 words). """
+    def __init__(self, total_flips: int, current_set: int = 1, bits_collected: str = "", mnemonic: list = None):
+        super().__init__()
+        self.total_flips = total_flips
+        self.current_set = current_set
+        self.bits_collected = bits_collected
+        self.mnemonic = mnemonic if mnemonic is not None else []
+        self.total_sets = 11 if total_flips == 128 else 23
+        self.last_set_bits = 7 if total_flips == 128 else 3
+
+    def run(self):
+        wordlist_language_code = self.settings.get_value(SettingsConstants.SETTING__WORDLIST_LANGUAGE)
+        required_bits = self.last_set_bits if self.current_set == self.total_sets + 1 else 11
+
+        ret = self.run_screen(
+            ToolsCoinEntropySetwiseEntryScreen,
+            current_set=self.current_set,
+            total_sets=self.total_sets + 1,
+            required_bits=required_bits
+        )
+
+        if ret == RET_CODE__BACK_BUTTON:
+            return Destination(ToolsCoinInputMethodView, view_args={"total_flips": self.total_flips})
+
+        self.bits_collected += ret
+
+        if self.current_set <= self.total_sets:
+            # Convert the 11-bit set to a BIP-39 word
+            word_index = int(ret, 2)
+            word = Seed.get_wordlist(wordlist_language_code)[word_index]
```

Comment:
This should be using the (currently unused) `get_bip39_word` method. And then that method should have some unit tests.